src/extractEnglishPages.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Created by Anshul on 3/6/2017.
+ */
+
+const path = require('path');
+const fs = require('fs');
+var franc = require('franc');
+const WARCStream = require('warc');
+var writeStream = fs.createWriteStream('./data/english.wet', { flags : 'w' });
+
+// extracting only english pages
+function extractEnglishPages(filepath : string) : void {
+
+    //choose wet files and other common crawl formats
+    if (path.extname(filepath).match(/\.wet|\.wat|\.warc/)){
+
+        // open each file in the folder as stream and pipe it to the warc parser
+        const WARCParser = new WARCStream();
+        fs.createReadStream(filepath).pipe(WARCParser).on('data', data => {
+
+            // write only english content to a new file
+            const content: string = data.content.toString('utf8');
+            var testString = content.substring(0,500);
+            if(franc(testString).match("eng")){
+                writeStream.write(content);
```

Comment:
Instead of only writing the content of a website in the file, it would be nice if we had meta data like the URL of the page. The best solution would probably be adopting the .wet file structure.